src/sentry/static/sentry/app/components/globalSelectionHeaderRow.jsx

```diff
@@ -11,21 +11,31 @@ class GlobalSelectionHeaderRow extends React.Component {
     multi: PropTypes.bool,
     onCheckClick: PropTypes.func.isRequired,
     children: PropTypes.node.isRequired,
+
+    /**
+     * This is a render prop which may be used to augment the checkbox rendered
+     * to the right of the row. It will receive the default `checkbox` as a
+     * prop anlong with the `checked` boolean.
+     */
+    renderCheckbox: PropTypes.func,
   };
 
   static defaultProps = {
+    renderCheckbox: ({checkbox}) => checkbox,
     multi: true,
   };
 
   render() {
-    const {checked, onCheckClick, multi, children, ...props} = this.props;
+    const {checked, onCheckClick, multi, renderCheckbox, children, ...props} = this.props;
+
+    const checkbox = <CheckboxFancy disabled={!multi} checked={multi && checked} />;
 
     return (
       <Container isMulti={multi} isChecked={checked} {...props}>
         <Content multi={multi}>{children}</Content>
-        <CheckboxWrapper onClick={multi ? onCheckClick : null} checked={checked}>
-          <CheckboxFancy disabled={!multi} checked={multi && checked} />
-        </CheckboxWrapper>
+        <CheckboxHitbox onClick={multi ? onCheckClick : null} checked={checked}>
+          {renderCheckbox({checkbox, checked})}
```

Comment:
would it help if it was `Checkbox` to indicate it's a react element?

Reply:
It's an instance of the already rendered checkbox, so it should not be uppercase. I considered turning it into a component, but decided this was fine just since we don't need more items in the tree.